### PR TITLE
Add stories for source academy

### PR DESCRIPTION
**Pull request by jflim98**
### Description

Stories for Source Academy is a lightweight technical blogging platform with a key focus on interactivity. It is inspired by the notebook interface which allows readers to easily evaluate code snippets within the blog. There is also a built-in editor for authors to easily and quickly write highly interactive technical blogs. Blogs are saved in a specific GitHub repository of the author in Markdown format.

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update
- [ ] Code quality improvements

### Checklist

- [x] I have tested this code
- [ ] I have updated the documentation

### CODE REVIEW

File: src/features/stories/storiesComponents/SourceBlock.tsx [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+      // Enable Subst Visualizer only for default Source 1 & 2
+      tabs.push({
+        label: 'Stepper',
+        iconName: IconNames.FLOW_REVIEW,
+        body: <SideContentSubstVisualizer content={processStepperOutput(output)} />,
+        id: SideContentType.substVisualizer
+      });
+    }
+
+    return tabs;
```
Comment (shenyih0ng) [inline src/features/stories/storiesComponents/SourceBlock.tsx:197]: awesome that story blocks have access to debugger tools like the playground! would there be future plans to add breakpoint support to story blocks for env viz?

also would be great if there can be a simple demo video to show the different functionalities and how we can use these debugger tools in stories! _right now i can't seem to get the stepper (added a video for ref below) and data viz to work but i have a feeling i'm doing something wrong_

https://user-images.githubusercontent.com/47914880/230731477-d3a9f993-3bb9-4be1-abff-644ce60c5399.mov

**Review by shenyih0ng — COMMENTED** [review]
mainly reviewed the stories functionality from a user's pov, will get more into the implementation details subsequently

**Comment by RichDom2185** [comment]
Helped to pull and resolve merge conflicts relating to the navigation bar refactor, as I reviewed the PR, so it would be better and faster.

But still left some things to do:

- [ ] Fix types to React 18 (e.g. `useCallback` parameters)
- [ ] Test files are still using Enzyme
- [ ] Snapshots might be weird (might need to check again after fixing tests)

**Review by RichDom2185 — APPROVED** [review]
As discussed offline, we will merge this PR first to prevent it from becoming too overly huge.

We hide the story routes/navigation bar button for now while it is still in development. Future PRs will address the issues incrementally.

**Pull request merged**